### PR TITLE
OFFAPPS-988 New translation strings for results count

### PR DESCRIPTION
**Pull request by yu-tian113**
Add new translation strings, update code to load corresponding results count translation

### CODE REVIEW

File: src/translations/en.yml [hunk 1]
```diff
@@ -93,6 +93,18 @@ Refresh your page for the changes to take effect and your Sidebar Search applica
       key: "txt.apps.search_app.search.results"
       title: "number of results the placeholder here will show something like 7 results"
       value: "{{count}} result(s):"
+  - translation:
+      key: "txt.apps.search_app.search.results_count.zero"
+      title: "Label indicating the number of results, when there are 0 results."
+      value: "{{count}} results"
```
Comment (vaevictis) [inline src/translations/en.yml]: I got bitten by that one before, but you need to create a screenshot where the translation is used, upload it to google drive and attach the link _within_ the translations file, as per [this documentation](https://zendesk.atlassian.net/wiki/spaces/i18n/pages/401277972/Upload+screenshots+to+Google+Drive).

File: src/javascript/modules/search.js [hunk 2]
```diff
@@ -248,6 +248,20 @@ class Search {
     this._render('.results-wrapper', getResultsTemplate)
   }
 
+  /**
+   * Get the translation key according to the number of search results
+   * @param {Number} count Number of the search results
+   */
+  _getResultsCountKey (count) {
```
Comment (vaevictis) [inline src/javascript/modules/search.js]: I've been pinged by @hiroiguchi about having any code in translation-focused PRs, but in the case of a block that is purposefully built to drive translations, I don't know? ❓

File: src/translations/en.yml [hunk 3]
```diff
@@ -93,6 +93,18 @@ Refresh your page for the changes to take effect and your Sidebar Search applica
...
+  - translation:
+      key: "txt.apps.search_app.search.results_count.one"
+      title: "Label indicating the number of results, when there are 1 result."
```
Comment (vaevictis) [inline src/translations/en.yml]: "there *are* 1 result"? wouldn't that be "there *is* 1 result"? or even "there *is* one result"?

File: src/javascript/modules/search.js [hunk 4]
```diff
@@ -248,6 +248,20 @@ class Search {
...
+    let key
+    switch (count) {
+      case 0: key = 'zero'; break
+      case 1: key = 'one'; break
+      default: key = 'other'
+    }
+    return `search.results_count.${key}`
+  }
```
Comment (vaevictis) [inline src/javascript/modules/search.js]: Also, maybe a test for this little utility function?

File: src/javascript/modules/search.js (hunk 2 above)
Comment (hiroiguchi) [reply src/javascript/modules/search.js]: @vaevictis Our team encourage Dev team to have separate PRs, one for code, one for strings which is an yml file. In other words, our team review only yml files and we need only yml files to translate. Although above code is related to strings but we don't review it. Thank you for pinging me!

**Comment by yu-tian113** [comment]
Close this PR, merged into tony/OFFAPPS-987

**Pull request closed without merging**